From 13f72756da86f155898e2c2022f7b3a106c3742e Mon Sep 17 00:00:00 2001
From: Linn Crosetto <linn@hp.com>
Date: Tue, 23 Apr 2013 12:26:45 -0600
Subject: x86/iommu/dmar: Remove warning for HPET scope type
Git-commit: 13f72756da86f155898e2c2022f7b3a106c3742e
Patch-mainline: v3.10-rc1
References: bnc#816969

ACPI_DMAR_SCOPE_TYPE_HPET is parsed by
ir_parse_ioapic_hpet_scope() and should not be flagged as an
unsupported type.

Signed-off-by: Linn Crosetto <linn@hp.com>
Cc: joro@8bytes.org
Cc: ddutile@redhat.com
Cc: suresh.b.siddha@intel.com
Link: http://lkml.kernel.org/r/1366741605-71293-1-git-send-email-linn@hp.com
Signed-off-by: Ingo Molnar <mingo@kernel.org>
Acked-by: Borislav Petkov <bp@suse.de>
---
 drivers/pci/dmar.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/pci/dmar.c b/drivers/pci/dmar.c
index c4a0235a4fdb..d753f4e692fc 100644
--- a/drivers/pci/dmar.c
+++ b/drivers/pci/dmar.c
@@ -132,7 +132,8 @@ static int __init dmar_parse_dev_scope(void *start, void *end, int *cnt,
 		if (scope->entry_type == ACPI_DMAR_SCOPE_TYPE_ENDPOINT ||
 		    scope->entry_type == ACPI_DMAR_SCOPE_TYPE_BRIDGE)
 			(*cnt)++;
-		else if (scope->entry_type != ACPI_DMAR_SCOPE_TYPE_IOAPIC) {
+		else if (scope->entry_type != ACPI_DMAR_SCOPE_TYPE_IOAPIC &&
+			 scope->entry_type != ACPI_DMAR_SCOPE_TYPE_HPET) {
 			printk(KERN_WARNING PREFIX
 			       "Unsupported device scope\n");
 		}
